src/dotnet-interactive-vscode/insiders/src/notebookControllers.ts

```diff
@@ -149,14 +144,17 @@ export class DotNetNotebookKernel {
                     diagnosticCollection.set(cell.document.uri, diags.filter(d => d.severity !== contracts.DiagnosticSeverity.Hidden).map(vscodeUtilities.toVsCodeDiagnostic));
                 }
 
-                return client.execute(source, getSimpleLanguage(cell.document.languageId), outputObserver, diagnosticObserver, { id: cell.document.uri.toString() }).then(() => {
+                return client.execute(source, getSimpleLanguage(cell.document.languageId), outputObserver, diagnosticObserver, { id: cell.document.uri.toString() }).then(async () => {
+                    await outputUpdatePromise;
```

Comment:
This `await` and the one a few lines below should be wrapped in a `try`/`catch` to ensure we always call `endExecution()` so the notebook doesn't look like it hung.